source/common/upstream/load_balancer_impl.cc

```diff
@@ -452,7 +462,11 @@ ZoneAwareLoadBalancerBase::ZoneAwareLoadBalancerBase(
           ASSERT(priority == 0);
           // If the set of local Envoys changes, regenerate routing for P=0 as it does priority
           // based routing.
-          regenerateLocalityRoutingStructures();
+          if (use_new_locality_routing_) {
+            regenerateLocalityRoutingStructures();
+          } else {
+            regenerateLocalityRoutingStructuresLegacy();
+          }
```

Comment:
Super nit (sorry): Can you leave the existing code alone and then add the new code as `regenerateLocalityRoutingStructuresNew` or whatever? I think this will make the diff show no changes for the old path and only new changes. Then we can swap the names when we clean up the runtime flag later? Thank you.

/wait

Reply:
done